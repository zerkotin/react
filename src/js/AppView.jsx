import React from 'react';
import {Route, Switch, withRouter, Link} from 'react-router-dom';
import PropTypes from 'prop-types';

import {Icon} from './components/components.jsx';

import '../style/app-view.less';

//TODO add example for redux
//TODO use lodash
/**
JSDOC example
AppView is the main view class for the application
*/
class AppView extends React.Component {

  constructor(props) {
    super(props);

    this.state = {path: null};
  }

  componentDidMount() {
    this.setState({path: this.props.location.pathname});
    this.props.history.listen(location => this.setState({path: location.pathname}));
  }

  render() {
    return (
      <div className="app-view">
        <HeaderView path={this.state.path}/>
        <div className="center-view">
          <NavigationView />
          <div id="main-view" className="main-view">
            <Switch>
              <Route exact path="/" component={HomeView}/>
              <Route path="/about" component={AboutView}/>
            </Switch>
          </div>
        </div>
        <FooterView />
      </div>
    );
  }
}

function HomeView() {
  return <div>Home View</div>;
}

function AboutView() {
  return <div>About View</div>;
}

function NavigationView() {
  const items = [
    <Link key="home" to="/"><Icon name="home"/>Home</Link>,
    <Link key="about" to="/about"><Icon name="info"/>About</Link>
  ];
  return <div className="navigation-view">{items}</div>;
}

function HeaderView({path}) {
  return <div className="header-view"><Icon name="settings"/>Path: {path}</div>;
}

HeaderView.propTypes = {path: PropTypes.string};

function FooterView() {
  return <div className="footer-view">Footer<Icon name="copyright"/></div>;
}

export default withRouter(AppView);
