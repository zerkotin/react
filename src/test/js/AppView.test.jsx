import React from 'react';
import renderer from 'react-test-renderer';
import {HashRouter, Route, Switch} from 'react-router-dom';

// import axios from 'axios';
// import MockAdapter from 'axios-mock-adapter';

import AppView from '../../js/AppView.jsx';

test('AppView', () => {
  // localStorage.user = '{"username": "appview"}'; //localStorage user mock

  // var mock = new MockAdapter(axios);
  // mock.onGet('/users/name/appview').reply(200, {fullName: 'AppView Test'}); //mock user request

  const component = renderer.create(
	<HashRouter>
		<Switch>
			<Route path="/" component={AppView}/>
		</Switch>
	</HashRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
