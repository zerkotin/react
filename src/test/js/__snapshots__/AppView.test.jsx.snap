// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView 1`] = `
<div
  className="app-view"
>
  <div
    className="header-view"
  >
    <i
      className="fas fa-settings"
    />
    Path: 
    /
  </div>
  <div
    className="center-view"
  >
    <div
      className="navigation-view"
    >
      <a
        href="#/"
        onClick={[Function]}
      >
        <i
          className="fas fa-home"
        />
        Home
      </a>
      <a
        href="#/about"
        onClick={[Function]}
      >
        <i
          className="fas fa-info"
        />
        About
      </a>
    </div>
    <div
      className="main-view"
      id="main-view"
    >
      <div>
        Home View
      </div>
    </div>
  </div>
  <div
    className="footer-view"
  >
    Footer
    <i
      className="fas fa-copyright"
    />
  </div>
</div>
`;
